docs/source/developers/release.rst

```diff
@@ -87,12 +87,49 @@ These are the different steps that are required to create a release candidate.
     # Curate the release
     # The end of the generated report shows the jira tickets with wrong version number assigned.
     archery release curate <version>
-    
+
+
+Creating a Release Candidate
+============================
+
+These are the different steps that are required to create a release candidate.
+
+The initial Release Candidate branch is merged from master. Follow up Release
+Candidates will be created from a maintenance branch.
+
+We have implemented a Feature Freeze policy between Release Candidates.
+This means that, in general, we will only add fixes between Release Candidates.
+Only specific features, if there is community consensus, will be merged to further Release Candidates.
```

Comment:
I think this needs rephrasing to make clear that this will only happen in special circumstances so as not to leave an "opening" for noncritical feature additions. Maybe:
```suggestion
In rare cases, critical features can be added between release candidates, if  there is community consensus.
```